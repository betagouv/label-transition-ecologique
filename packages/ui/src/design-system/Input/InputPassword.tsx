import {Ref, forwardRef, useState} from 'react';
import {InputBase, InputBaseProps} from './InputBase';

export type InputPasswordProps = Omit<InputBaseProps, 'icon' | 'type'>;

/**
 * Affiche un champ de saisie de mot de passe.
 * Un bouton permet d'afficher/masquer le contenu du champ.
 */
export const InputPassword = forwardRef(
  ({...remainingProps}: InputPasswordProps, ref?: Ref<HTMLInputElement>) => {
    const [passwordVisible, setPasswordVisible] = useState(false);

    return (
      <InputBase
        type={passwordVisible ? 'text' : 'password'}
        icon={{
          buttonProps: {
            type: 'button',
            icon: passwordVisible ? 'lock-password-line' : 'eye-line',
            title: passwordVisible
              ? 'Cacher le mot de passe'
              : 'Voir le mot de passe',
            onClick: e => {
              e.preventDefault();
              setPasswordVisible(!passwordVisible);
            },
          },
        }}
        ref={ref}
        {...remainingProps}
      />
    );
  }
);
