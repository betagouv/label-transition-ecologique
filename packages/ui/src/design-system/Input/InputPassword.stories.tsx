import {Meta, StoryObj} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import {InputPassword} from './InputPassword';

const meta: Meta<typeof InputPassword> = {
  title: 'Design System/InputPassword',
  component: InputPassword,
};

export default meta;

type Story = StoryObj<typeof InputPassword>;

/** Sans aucune props renseignée. */
export const Default: Story = {
  args: {},
};

/** Avec une valeur renseignée. */
export const AvecValeur: Story = {
  args: {
    value: 'hyper secret',
    onChange: action('onChange'),
  },
};
