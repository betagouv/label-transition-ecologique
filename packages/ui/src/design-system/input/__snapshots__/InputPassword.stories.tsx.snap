// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/InputPassword AvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="password"
         class="text-grey-8 px-4 py-3 outline-none text-md border-r border-solid border-grey-4"
         value="hyper secret"
  >
  <button title="Voir le mot de passe"
          class="rounded-lg w-fit border border-solid font-bold group cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-eye-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/InputPassword Default smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="password"
         class="text-grey-8 px-4 py-3 outline-none text-md border-r border-solid border-grey-4"
  >
  <button title="Voir le mot de passe"
          class="rounded-lg w-fit border border-solid font-bold group cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-eye-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;
